### DISCUSSION TITLE
crazyflie m2 led light remains solid blue

### DISCUSSION THREAD

**Discussion by rajatag77** [bitcraze/discussions#172] [Q&A] [answered]
Hi, when I turn on my crazyflie only M2 lights up solid. If I try to put it on bootloader mode, only m2 blinks. 
Only the M2 Led is fully lit (Blue) when I plug the USB cable.
With battery plug-in, same result: only M2 blue Led fully lit.
I can't charge the battery.
I have several batteries (crazyFlie2 batteries) and tested with different ones: the same result.
I also tested several USB cables, without success.
Board is clean. No big crash or shock on the board...

I was working on Crazyswarm. It worked for a few runs then this problem occurred.
VCC measured is 3V.
Also, I have started the board in bootloader mode by holding the power button and starting it for 3 seconds until the M2 led starts to blink to check if M3 led starts to blink then. No luck...

1. The VCC is at 2.99V so that's not the problem.
2. I tried using the cfclient to reset the bootloader using the instructions (replug radio, etc) and nothing happens.
3. I have changed out the batteries and left the crazyflie unplugged for a few minutes.
4. I tried powering it just over USB and there is no change.
5. I tried using the cfloader flash tool but it says it can not connect to the bootloader.

I also tried cold bootloader mode with USB as well as Batteries.

**Comment by knmcguire** [comment]
Hi! 

There are several things that can be wrong:

1. I see, also in other duplicate post that you might haven't tried everything yet We have this troubling shooting guide that you could go through again: https://www.bitcraze.io/support/troubleshooting/#crazyflie-is-bricked (edited), especially the forcing into bootloader mode.

Other than what is described there, there might be two things that are causing this as well:

2. The wrong firmware was flashed on the crazyflie. If you would flash firmware meant for the 'bolt' or the 'roadrunner', the crazyflie will not startup properly
3. The STM chip itself is broken, which you can notice if it heats up significantly when attaching a battery that it is almost too hot to the 

Mind that we recently switched to this support platform so there might be other things to try out that people have posted on the old forum: https://forum.bitcraze.io/

**Reply by rajatag77** [reply]
Thanks for your reply.

1. I have tried all the parameters that mentioned in the troubleshooting guide but nothing worked. 

2. We are using lighthouse position system with 4 base stations. In order to use we need to change no of base stations to 4 in bitcraze recent firmware, we need to run "make menuconfig" and go to the Expansion deck configuration menu and set Max number of base stations to "4". 
So for your second point we are recent firmware and not flashing for 'bolt' or 'roadrunner' but for ROS-based Crazyswarm.

3. None of the crazyflies STM chip is broken or heating up when attaching to the battery.

I am waiting for any other measures that might help me in this matter.

**Reply by knmcguire** [reply]
Ah woops sorry! For point 2 I had copy pasted the wrong link. This is the right trouble shooting guide https://www.bitcraze.io/support/troubleshooting/#crazyflie-is-bricked. Could you take a look there?

I've updated the link above as well.

**Reply by rajatag77** [reply]
sorry for the inconvinience.
I have tried all the points including recovery mode but still, nothing happened the same problem exists. I have shared the video of recovery mode as well.

https://user-images.githubusercontent.com/77656817/183038039-dbd649f9-045b-4e75-9013-ae7307160aab.mp4

**Reply by knmcguire** [reply]
Hmm that is strange. The status is supposed to show
` Firmware (1/2) Uploading buffer to STM32`
but it goes to finished right away. So that is very strange indeed.

Could you copy the content from  the cfclient's 'help'->'about'->'debug tab' So that we know everything from your cfclient and python

**Reply by rajatag77** [reply]
Cfclient
Cfclient version: 2022.3
System: linux
Python: 3.9.7
Qt: 5.15.2
PyQt: 5.15.6

Interface status
radio: Crazyradio version 0.53
UsbCdc: No information available
udp: None
prrt: No information available

Input readers
linuxjsdev (0 devices connected)

Input devices
None

Crazyflie
Connected: None
Firmware: None

Decks found

Sensors found

Sensors tests

**Comment by knmcguire** [comment]
So I'm a bit worried about the process of how the cfclient looks like during flashing of coldboot...

I see that you have an older version of theh cfclient, so could you perhaps do:

```
pip3 uninstall cfclient cflib
pip3 install cfclient
```
and try to coldboot again?

**Reply by rajatag77** [reply]
still not working
attaching the video of that

https://user-images.githubusercontent.com/77656817/183411558-ebf24f7a-52db-4f5a-bfd6-35f74f843803.mp4

**Reply by rajatag77** [reply]
Updated Cfclient debug file with one of new crazyflie (in working condition), lighthouse deck attached

Cfclient
Cfclient version: 2022.5
System: linux
Python: 3.9.12
Qt: 5.15.2
PyQt: 5.15.7

Interface status
radio: Crazyradio version 0.53
UsbCdc: No information available
udp: None
prrt: No information available

Input readers
linuxjsdev (0 devices connected)

Input devices
None

Crazyflie
Connected: radio://0/125/2M/E7E7E7E707
Firmware: 46f497a1ef05 (CLEAN)

Decks found
bcLighthouse4: rev=D, adr=0D1B59B600000065

Sensors found
BMP388: 1
AK8963: 0
LPS25H: 0

Sensors tests
MPU6500: 0
AK8963: 0
LPS25H: 0

**Comment by knmcguire** [comment, marked as answer]
So I've been working on recreating your problem based on the videos and cfclient. We believe that the symptoms are consistent with a broken STM bootloader, and I verified by wrongly flashing mine and I saw the same cfclient behavior when updating the firmware.

Edit: The terminal also said `Bootloader protocol oxFF not supported` when it failed to flash which gave it away too. 

Did you by any time tried to do a DFU update of the crazyflie with the micro usb? It could be that you have flashed over the stm bootloader and removed the over air flashing capabilities.

I think you can fix your crazyflie via these steps:

1. Recover the stm bootloader via a usb cable and dfu update via these instructions: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/dfu/#bootloader-recovery. 
2. After this, you should be able to put the crazyflie in a proper bootloader mode where both the M2 and M3 are flashing (just release the button quickly once m2 starts blinking, since holding it longer than 5 seconds will put it in a different mode).
3. Flash the latest release by cold boot recovery of the crazyflie client. You should see this status at the start:

![image](https://user-images.githubusercontent.com/11061260/183607354-a662aa3f-ce31-4f45-9ccc-a6f12aa6e5b7.png)

Let us know how this went.

**Reply by rajatag77** [reply]
It worked!
Thanks a lot for your patience and for guiding me through this procedure.

**Reply by knmcguire** [reply]
Ah great! thanks. You too for providing information about your problem. It was difficult to determine what the issue was initially but the videos and debug information helped!

**Reply by knmcguire** [reply]
btw, now the overairflashing is working try to not use the dfu-update with the usb cable instructions again and only use over-air flashing. The dfu update method is too sensitive for breaking the stm bootloader, and we hardly hear it happening if only the crazyradio is used for the firmware update. That is why we only advise to use it as recovery method.

**Reply by rajatag77** [reply]
Sure, we will keep this in mind. Also, just to mention what I think might have caused the issue in the first place: I had updated the crazyflie with the latest available at [https://github.com/bitcraze/crazyflie-firmware](url). I used `make cload` to flash the firmware rather than using cfclient. That is because I use 4 light house base stations. After that, I tried flying the crazyflie using crazyswarm and that's when the error occurred. This happened with multiple of our crazyflies until I reverted back to an older version of the firmware. So I'd like to request you to kindly look into this as well since this might be useful for other users as well.

**Reply by matejkarasek** [reply]
One of our Flapper users had a similar issue with the Bolt 1.1, this solution worked to fix it.

Interestingly, also they reported that the issue appeared when they tried connecting to 2 units at a time to fly them simultaneously (the units operated normally before), which caused the Bolt to go to the 1 solid blue led state. It may be, the units had the same, default address when they were connecting to them. So maybe something to look into?

**Comment by Drondelbee** [comment]
Hey i have an similar problem. I have assembled all parts except the flow deck and AI bundle. At this point, the LED lights all worked fine as per the functional check of the instructions with propeller movement as well. I then downloaded the Iphone Mobile Application and a flight start would have tended to be possible. After that I wanted to update the firmware before mounting the flow deck according to the instructions. When installing up to the 100% everything worked and at the end an error was displayed. Since then, the start initiation no longer works and only LED light M2 is in continuous operation. Through the power button, the light can be turned off or also lead by holding down in the flashing mode. I have Windows10 as a system and have meanwhile run the cfclient via Python and tried to connect with the Crazyradio2.0. Even with the coldboot it did not lead to the goal. 

Now I am stumped and ask for technical support....

![IMG_9829](https://github.com/bitcraze/discussions/assets/74738815/5975d851-ab8c-4061-bb8d-f5370a1ef444)
![IMG_9836](https://github.com/bitcraze/discussions/assets/74738815/a34d23f8-beb1-4571-8413-ebf45c0a374d)
sets/74738815/a2f7f974-804f-4493-ab47-a74dc8557261)
![IMG_9835](https://github.com/bitcraze/discussions/assets/74738815/1015d20b-b16e-45cf-b7ad-0c9b2f190080)

**Reply by tobbeanton** [reply]
How did you do the cold boot? Did you do it from an un-powered state, that is: holding the button while connecting the battery?

**Reply by Drondelbee** [reply]
Yes I did the cold boot from an un-powered state. Holding the power button while connecting the battery. Then the m2 LED is flashing but I don´t get a connection then and the flashing blue light don´t stop until you disconnect the battery.

**Reply by robu** [reply]
I just want to confirm that I'm in the exact same situation. Build was fine, self-test worked. I installed the iOS app to test, and the first thing I did was an over-the-air update of firmware. Unfortunately, I now have a solid blue M2 led as the only visible sign anything at all is working. No self-test on startup. Oh, it seems that I can also connect via the iOS app, but there is nothing I can do to make the drone have any reaction from the app.

**Reply by tobbeanton** [reply]
That is very strange, "bricking" the STM32 should be practically impossible with the bootloader. Can you test if you can get the STM32 in DFU update mode like in [these](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/dfu/#bootloader-recovery) instructions?

**Reply by robu** [reply]
The only additional clue I can provide is that the OTA update was stopped a couple of times due to the app losing focus on my iPhone (due to external factors), which caused the transfer to halt. I restarted manually, and then eventually I got it to finish (causing the situation I'm now in). I'd be happy to try the instructions you posted, but I don't have access to a linux computer right now. Is there a Mac version of the instructions I can try?

**Reply by Drondelbee** [reply]
Ok I got the fist steps. Forced the crazyflie into the DFU-Mode and connected with the USB-cable and disconnected battery. I can find the crazyflie as STM32 BOOTLOADER on my usb-Port (all good with Drivers). I downloaded the cf2loader-1.0.bin but in my linux window isn´t shown the STM32... with "lsusb" command. I get just these two results:

lsusb

>  Bus 002 Device 001: ID XXXX:XXX3 Linux Foundation 3.0 root hub
> Bus 001 Device 001: ID XXXX:XXX2 Linux Foundation 2.0 root hub

But the "Bus XXX Device XXX: ID 0483:df11 STMicroelectronics STM Device in DFU Mode" isn´t shown in the linux window...?

And with the command 

make flash_dfu_manual

> *** No rule to make target 'flash_dfu_manual'.   Stop.

But in the device-manager under USB-Devices the STM32 BOOTLOADER is shown. I can see in zadig-2.8 the STM32BOOTLOADER too when the crazyflie is connected over USB-connection but just in DFU-MOde logicially.

Did i Forgot to install something in linux?

**Reply by tobbeanton** [reply]
@robu dfu-util should be available for MAC as well, e.g. [here](https://formulae.brew.sh/formula/dfu-util). If you can get that running you should be able to do the same as in the linux instructions.

**Reply by robu** [reply]
> @robu dfu-util should be available for MAC as well, e.g. [here](https://formulae.brew.sh/formula/dfu-util). If you can get that running you should be able to do the same as in the linux instructions.

Installed dfu-util and managed to run the line in "Bootloader recovery" that you linked to. The command seemed to complete successfully, but when I remove the USB cable and reconnect the battery, I'm back to where I started (solid blue M2 LED).

This was the output from running the dfu-util command:
```
dfu-util 0.11

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2021 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

dfu-util: Warning: Invalid DFU suffix signature
dfu-util: A valid DFU suffix will be required in a future dfu-util release
Opening DFU capable USB device...
Device ID 0483:df11
Device DFU version 011a
Claiming USB DFU Interface...
Setting Alternate Interface #0 ...
Determining device status...
DFU state(10) = dfuERROR, status(10) = Device's firmware is corrupt. It cannot return to run-time (non-DFU) operations
Clearing status
Determining device status...
DFU state(2) = dfuIDLE, status(0) = No error condition is present
DFU mode device DFU version 011a
Device returned transfer size 2048
DfuSe interface name: "Internal Flash  "
Downloading element to address = 0x08000000, size = 7192
Erase   	[=========================] 100%         7192 bytes
Erase    done.
Download	[=========================] 100%         7192 bytes
Download done.
File downloaded successfully
```

**Reply by robu** [reply]
I re-did the complete process once more and had exactly the same result: same output from dfu-util, and ended up with the same error condition after.

**Reply by tobbeanton** [reply]
Now you have two options. Also flash the STM32 firmware with dfu-util or use the radio bootloader in cold boot mode. That is hold the button while connecting the battery.

**Reply by robu** [reply]
Sure, dfu-util was easy to handle, so I can try that. Can you point me to directions on how that dfu-util line should look, and where to get a stable firmware file?

**Reply by Drondelbee** [reply]
https://github.com/redbear/Duo/blob/master/docs/dfu-util_installation_guide.md

In this guide u can find the dfu-util install-guide.  The dfu-util line is here shown... 

https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/dfu/#bootloader-recovery

**Reply by robu** [reply]
Ok, @tobbeanton , I have now installed the toolchain needed and built a `cf2.bin` file. How can I use dfu-util to install it on the drone? I assume the parameters are different from when I installed the bootloader-recovery previously?

**Reply by tobbeanton** [reply]
The [DFU guide](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/dfu/) should work. Now when you can build the FW try: `make flash_dfu_manual` after putting the CF2-STM32 in dfu update mode.

**Reply by robu** [reply]
> The [DFU guide](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/dfu/) should work. Now when you can build the FW try: `make flash_dfu_manual` after putting the CF2-STM32 in dfu update mode.

Got it. Thank you! Now I'm back to the same seemingly working state as before I tried to update the firmware!

**Reply by Drondelbee** [reply]
@robu : I guess that you installed the Docker App like in the INstallation guide  for the bitcraze/toolbelt (https://hub.docker.com/r/bitcraze/toolbelt). Do you also Add a alias to your ./bashrc proile?  If yes can you answer it in this new discussion find here:  https://github.com/orgs/bitcraze/discussions/755

Thanks in advance.

**Reply by robu** [reply]
> @robu : I guess that you installed the Docker App like in the INstallation guide for the bitcraze/toolbelt (https://hub.docker.com/r/bitcraze/toolbelt). Do you also Add a alias to your ./bashrc proile? If yes can you answer it in this new discussion find here: https://github.com/orgs/bitcraze/discussions/755
> 
> Thanks in advance.

Sorry, but I didn't. I just followed the OSX instructions [here](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/building-and-flashing/build/), since I'm on a Mac, and got it running fairly easy. No Docker.

**Comment by Drondelbee** [comment]
I have installed the libusbK (v3.1.0.0) and tried the others. Now I´m working with Zadig2.8 and the WINUSB (v.6.1.7600.16385). 

When i put in the command:

sudo dfu-util -d 0483:df11 -a 0 -s 0x08000000 -D cf2loader-1.0.bin   (in linux window) I get the results below: 

dfu-util 0.9

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2016 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

dfu-util: Could not open file cf2loader-1.0.bin for reading: No such file or directory

**Reply by tobbeanton** [reply]
Looks like you are almost there!
> dfu-util: Could not open file cf2loader-1.0.bin for reading: No such file or directory

Is the file in the same directory as where you are running dfu-util from?

**Reply by Drondelbee** [reply]
The cf2loader-1.0.bin File I have on Desktop.

**Reply by Drondelbee** [reply]
The Dfu-util.exe file I put under 

:C/Programm Files/dfu-util/dfu-util.exe

**Reply by Drondelbee** [reply]
Also the command lsusb do Not show the STM32 Device in the linux window. Is that normal? But I can see ist in device manager as STM32BOOTLOADER

**Reply by tobbeanton** [reply]
It might be easier to try and use windows and the windows port of [dfu-util](https://dfu-util.sourceforge.net/)

**Reply by Drondelbee** [reply]
With the cmd "dfu-util -l" in Windows I get following result (Found but cannot open DFU device)... What do you mean with dfu-util port in windows? The cf2loader.bin file where I have to put it in (which Path) - I have located it in the path C:\Users\Drondelbee\AppData\Local\particle\bin\dfu-util 
And the file dfu-util contains :
![1](https://github.com/bitcraze/discussions/assets/74738815/5f20110f-0676-4c35-bb23-63de998be4b5)


dfu-util -l

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2014 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to dfu-util@lists.gnumonks.org

Cannot open DFU device 17e9:XXXX
Cannot open DFU device 17e9:XXXX
Found DFU: [0483:df11] ver=XXXX, devnum=XX, cfg=X, intf=0, alt=3, name="@Device Feature/0xFFFF0000/01*004 e", serial="XXXXXXXX"
Found DFU: [0483:df11] ver=XXXX, devnum=XX, cfg=X, intf=0, alt=2, name="@OTP Memory /0x1FFF7800/01*512 e,XXXXXX", serial="XXX"
Found DFU: [0483:df11] ver=2200, devnum=XX, cfg=X, intf=0, alt=1, name="@Option Bytes  /0x1FFFC000/01*016 e", serial="XXXXXXX"
Found DFU: [0483:df11] ver=2200, devnum=XX, cfg=X, intf=0, alt=0, name="@Internal Flash  /0x08000000/XX*XXXKg,01*064Kg,07*128Kg", serial="XXXXXXXXX"

**Reply by tobbeanton** [reply]
OK so you are already running the windows flavor of dfu-util or what do you mean with (in linux window)? 
Looks like it found the right device ` Found DFU: [0483:df11]` but still you get:
`dfu-util: Could not open file cf2loader-1.0.bin for reading: No such file or directory`?

**Reply by Drondelbee** [reply]
Windows Subsystem for Linux = linux Windows. Yes I still geh the Info no such file per directory… can it have to do, because of „Cannot Open the device STM32 But the Device is found…?“

**Reply by Drondelbee** [reply]
In conclusion I have one good result and more questions. First of I found why windows command "dfu-util --list" found DFU Device but the message can not open is now gone. I have disconnected my laptop from all extern connections and see there cannot open DFU-Device is gone.

Now I have questions:

1. First compile the firmware as normal with make-command, due to the bitcraze manual DFU Flashing Guide. I don´t get what I ´ve do for it?
(https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/dfu/)

2. How i can make_flash_dfu_manual  ?

3. This command "sudo dfu-util -d 0483:df11 -a 0 -s 0x08000000 -D cf2loader-1.0.bin" 

Then I get following result: "dfu-util: Could not open file cf2loader-1.0.bin for reading: No such file or directory"

I don´t know what is the Reason. Can it have to do with the UDEV-Rules? If yes where can I find them and how i flash them?

Thanks in advance!

**Reply by krichardsson** [reply]
Hi,
these questions are too far from the title of the topic "crazyflie m2 led light remains solid blue", please start a new discussion
Thanks!

**Reply by Drondelbee** [reply]
Ok, I ´ve forgot something in the toolchain. I ´ll handle the problem, then I will see what happened.

**Reply by Drondelbee** [reply]
I´ve opened a new discussion for my question due to the INstallation of Docker bitcraze/toolbelt here:
 https://github.com/orgs/bitcraze/discussions/755
